swiss_ai/datasets/HDVILA.md

```diff
@@ -0,0 +1,65 @@
+# [HDVILA-100M](https://github.com/microsoft/XPretrain/tree/main/hd-vila-100m)
+Page adapted from [dataset_examples/HDVILA.md](/dataset_examples/HDVILA.md).
+
+HDVILA 100M is a dataset of 3.3M high-resolution videos from YouTube, divided into 100M clips.
+
+## Download the metdata
+First, run `wget -O hdvila100m.zip https://hdvila.blob.core.windows.net/dataset/hdvila100m.zip?sp=r&st=2022-06-28T03:33:11Z&se=2026-01-01T11:33:11Z&spr=https&sv=2021-06-08&sr=b&sig=VaqQkLFDqKinfkaPNs1jJ1EQIYCB%2FUPYiqFqmjWye6Y%3D` to download the HD VILA 100M metadata. Next, just run `unzip hdvilla100m.zip` in order to unzip the metadata. You should now have an `hdvila100m/` directory.
+
+Next, we need to do some preprocessing to get this metadata formatted into a nice parquet. The following script will take the downloaded metadata `.jsonl` files and create a parquet with all the relevant information.
+
+```python
+import pandas as pd
+import glob
+import json
+import os
+import time
+from datetime import datetime
+
+def time_string_to_seconds(timestamp):
+    hh,mm,s = timestamp.split(':')
+    ss,ms = s.split('.')
+    time = 3600*int(hh) +  60*int(mm) + int(ss) + int(ms)/1000
+    return time
+
+def convert_clip_list(clip_list):
+    return [[time_string_to_seconds(x) for x in clip] for clip in clip_list]
+
+parquet_dir = "/path/to/my/metadata/dir/"
+
+data = []
+for jsonl in sorted(glob.glob(f"{parquet_dir}*.jsonl")):
+    path = os.path.join(parquet_dir, jsonl)
+    with open(path, "r") as f:
+        for line in f:
+            json_obj = json.loads(line)
+            clips = [
+                json_obj['clip'][i]['span']
+                for i in range(len(json_obj['clip']))
+            ]
+
+            out = {
+                'video_id': json_obj['video_id'],
+                'url': json_obj['url'],
+                'clips': clips
+            }
+            data.append(out)
+
+df = pd.DataFrame(data)
+df['clips'] = df['clips'].map(lambda x: convert_clip_list(x))
+df.to_parquet("hd_vila.parquet")
+```
+
+Once you run this, you should have a file `hd_vila.parquet` with all the relevant metadata.
+
+## Download the Videos
+To download the videos on clariden, just run video2dataset with the [default download config for clariden](../configs/download_clariden.yaml): From the login node, execute the following command, adapting the paths if necessary
+
+```
+video2dataset --url_list="hd_vila.parquet" --config="swissai/configs/download_clariden.yaml" --output_folder="./hd_vila_v2d" --input_format="parquet" --output_format="webdataset" --url_col="url" --encode_formats="{'video': 'mp4', 'audio':'m4a'}" 
```

Comment:
small quick thing: `config=swiss_ai/...` right?

Reply:
Also I kinda like the idea of having all of these just be a script? Like input zip file/output dir and then the script constructs the v2d dataset. If it's easy to do could you maybe port some of this into a `utils/process_hdvila.py`? Feel free to push back though if you feel like this is unnecessary (I agree it is to an extent, but also it might be worth it for cleanliness/consistency going forward if the cost now is low enough).

Reply:
> small quick thing: `config=swiss_ai/...` right?

fixed, thanks

Reply:
To the script: I feel like it's not worth it. It will only be run maybe once or twice more, so as long as there's clear instructions I think it's fine. Especially because every time we run it again is probably gonna be on a slightly different system where things might have to be adapted anyways. And some of the parts take a few hours (e.g. the downloading of MiT, its processing, or all video2dataset downloads), so it might be nice to run them one by one anyways and monitor manually